assembl/static2/js/app/components/common/avatar.jsx

```diff
@@ -2,34 +2,35 @@ import React from 'react';
 import { connect } from 'react-redux';
 import { Translate } from 'react-redux-i18n';
 import { Link } from 'react-router';
-import Loader from './loader';
+import { LinkContainer } from 'react-router-bootstrap';
+import { DropdownButton, MenuItem } from 'react-bootstrap';
+import { Routes } from '../../routes';
 
 class ProfileIcon extends React.Component {
   render() {
     const { debateData } = this.props.debate;
-    const { users, usersLoading, usersError } = this.props.users;
-    const { rootPath, connectedUserId } = this.props.context;
+    const slug = { slug: debateData.slug };
+    const { connectedUserId, connectedUserName } = this.props.context;
     return (
       <div className="right avatar">
         {!connectedUserId &&
-          <Link to={`${rootPath}${debateData.slug}/login`}>
+          <Link to={`${Routes.getContextual('login', slug)}?next=${Routes.get('home', slug)}`}>
```

Comment:
I've done that in most places on the new front-end. I might have missed a few, but I've already made most of the migration.